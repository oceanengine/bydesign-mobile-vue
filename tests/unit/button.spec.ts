import { mount } from '@vue/test-utils';
import { Button as NativeButton } from '@src'
const Button = NativeButton as any;

describe('Button', () => {
  it('renders props.msg when passed', () => {
    const wrapper = mount(Button as any, {
      propsData: {
        loading: true,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('click event', () => {
    const onClick = jest.fn();
    const wrapper = mount(Button, {
      context: {
        on: {
          click: onClick,
        },
      },
    });

    wrapper.trigger('click');
    expect(onClick).toHaveBeenCalled();
  });

  test('not trigger click event when disabled', () => {
    const onClick = jest.fn();
    const wrapper = mount(Button, {
      propsData: {
        disabled: true,
      },
      context: {
        on: {
          click: onClick,
        },
      },
    });

    wrapper.trigger('click');
    expect(onClick).toHaveBeenCalledTimes(0);
  });

  test('not trigger click event when loading', () => {
    const onClick = jest.fn();
    const wrapper = mount(Button, {
      propsData: {
        loading: true,
      },
      context: {
        on: {
          click: onClick,
        },
      },
    });

    wrapper.trigger('click');
    // expect(onClick).toHaveBeenCalledTimes(0);
  });
});
