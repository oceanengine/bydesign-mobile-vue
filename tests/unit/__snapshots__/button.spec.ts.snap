// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders props.msg when passed 1`] = `
<div class="byted-button byted-button--default byted-button--normal">
  <div class="byted-loading byted-button__loading"><i class="byted-loading__icon byted-loading-circle">test-file-stub</i></div>
</div>
`;
